packages/shared-ux/error_boundary/src/services/error_boundary_services.tsx

```diff
@@ -8,39 +8,54 @@
 
 import React, { FC, useContext, useMemo } from 'react';
 
-import { KibanaErrorBoundaryServices } from '../../types';
+import { KibanaErrorBoundaryProviderDeps, KibanaErrorBoundaryServices } from '../../types';
 import { KibanaErrorService } from './error_service';
 
 const Context = React.createContext<KibanaErrorBoundaryServices | null>(null);
 
 /**
  * A Context Provider for Jest and Storybooks
+ * @internal
  */
 export const KibanaErrorBoundaryDepsProvider: FC<KibanaErrorBoundaryServices> = ({
   children,
   onClickRefresh,
   errorService,
-}) => {
-  return <Context.Provider value={{ onClickRefresh, errorService }}>{children}</Context.Provider>;
-};
+}) => (
+  <Context.Provider value={{ onClickRefresh, errorService, hasStatefulDeps: true }}>
+    {children}
+  </Context.Provider>
+);
 
 /**
- * Kibana-specific Provider that maps dependencies to services.
+ * Provider that uses dependencies to give context to the KibanaErrorBoundary component
+ * This provider is aware if it was already called in a higher level of the component tree
+ * @public
  */
-export const KibanaErrorBoundaryProvider: FC = ({ children }) => {
-  const value: KibanaErrorBoundaryServices = useMemo(
-    () => ({
+export const KibanaErrorBoundaryProvider: FC<KibanaErrorBoundaryProviderDeps> = ({
+  children,
+  ...deps
+}) => {
+  const parentContext = useContext(Context);
+  const value: KibanaErrorBoundaryServices = useMemo(() => {
+    if (parentContext && parentContext.hasStatefulDeps) {
```

Comment:
This PR sneaks in a small change: Reusing context that was already created if somehow, `KibanaErrorBoundaryProvider` is rendered within another `KibanaErrorBoundaryProvider`.